pkg/steps/new_e2e.go

```diff
@@ -139,12 +145,26 @@ func (s newE2ETestStep) genPods(steps []api.TestStep) ([]coreapi.Pod, error) {
 		if owner := s.jobSpec.Owner(); owner != nil {
 			pod.OwnerReferences = append(pod.OwnerReferences, *owner)
 		}
+		if step.ArtifactDir != "" {
+			if artifacts == nil {
```

Comment:
Having a possibly-nil worker is a bit awkward. Why not:

```go
type collectionRecord struct{ podName, containerName string }
var toCollect []collectionRecord

// ....

for _, step := range steps {
    if step.ArtifactDir != "" {
        // add the volume mount and container
        toCollect = append(toCollect, collectionRecord{podName: pod.Name, containerName: container.Name})
    }
}

if len(toCollect) > 0 {
    // create worker
    // return notifier
}
```

Reply:
We don't even need the `struct` because we know what `podName` will be based on `containerName`. I also want to check if we can use a single worker for all three phases.